Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -73,6 +73,26 @@ namespace
     using namespace AzToolsFramework::AssetSystem;
     using namespace AzFramework::AssetSystem;
 
+    template<typename T>
+    void BuildReport(AssetProcessor::ISourceFileRelocation* relocationInterface, T& result, AZStd::vector<AZStd::string>& lines)
+    {
+        if (result.IsSuccess())
+        {
+            AssetProcessor::RelocationSuccess success = result.TakeValue();
+
+            // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+            AZStd::string report = relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+            AzFramework::StringFunc::TokenizeVisitor(
+                report.c_str(),
```

Comment:
The `.c_str()` call is unnecessary here. `AZStd::string` has a conversion operator to `AZStd::string_view` which can construct a string_view by supplying the address to the string and it's length.
For a c-string, an [additional call to strlen](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/string_view.h#L645-L653) is needed to re-calculate the string length which is already known by the `AZStd::string`

Reply:
I'm still working in this area so will make the change in there.